pkg/helmdeployer/deployer_test.go

```diff
@@ -0,0 +1,60 @@
+package helmdeployer
+
+import (
+	"fmt"
+	"runtime"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestValuesFrom(t *testing.T) {
+	a := assert.New(t)
+	key := "values.yaml"
+	newline := "\n"
+	if runtime.GOOS == "windows" {
+		newline = "\r\n"
+	}
+
+	configMapPayload := fmt.Sprintf("replication: \"true\"%sreplicas: \"2\"%sserviceType: NodePort", newline, newline)
+	secretPayload := fmt.Sprintf("replication: \"false\"%sreplicas: \"3\"%sserviceType: NodePort%sfoo: bar", newline, newline, newline)
+	expected := map[string]interface{}{
+		"replicas":    "2",
+		"replication": "true",
+		"serviceType": "NodePort",
+		"foo":         "bar",
+	}
+
+	configMapName := "configmap-name"
+	configMapNamespace := "configmap-namespace"
+	configMapValues, err := processValuesFromObject(configMapName, configMapNamespace, key, nil, &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      configMapName,
+			Namespace: configMapNamespace,
+		},
+		Data: map[string]string{
+			key: configMapPayload,
+		},
+	})
+	a.NoError(err)
+
+	secretName := "secret-name"
+	secretNamespace := "secret-namespace"
+	secretValues, err := processValuesFromObject(secretName, secretNamespace, key, &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: secretNamespace,
+		},
+		Data: map[string][]byte{
+			key: []byte(secretPayload),
+		},
+	}, nil)
+	a.NoError(err)
+
+	totalValues := make(map[string]interface{})
+	totalValues = mergeValues(totalValues, secretValues)
+	totalValues = mergeValues(totalValues, configMapValues)
+	a.Equal(expected, totalValues)
```

Comment:
Changed since last review: I've also added `mergeValues` to the unit test for a more comprehensive test.